### DISCUSSION TITLE
Cannot uninstall Aspire Project Templates in Visual Studio

### DISCUSSION THREAD

**Discussion by snuup** [microsoft/aspire#6584] [General]
I do not remember how I installed them, I think it was something like an "Aspire SDK". However such thing apparently does not exist. I would be very happy if I could remove them from Visual Studio, tried for hours, all the methods mentioned anywhere. Can you explain which "installs" happen, I mean there is certainyl no Windows Program installed. I removed the Aspire Package with the VS Installer and cleared ProjectTemplate caches and ran devenv /updateconfiguration.

**Comment by davidfowl** [comment]
2 ways, aspire uses a .NET workload. You can use dotnet workload list to see this:

```
➜ dotnet workload list

Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
aspire                     8.2.2/8.0.100         VS 17.13.35430.269

Use `dotnet workload search` to find additional workloads to install.
```

Notice the install source says VS (this is because visual studio also installs this workload)

There's also a Visual studio workload (which installs the aspire dotnet workload):

Open up the visual studio installer:
1. Click Modify
2. Click individual components
3. Search for aspire

<img width="934" alt="image" src="https://github.com/user-attachments/assets/db248597-08cd-4591-b4a1-117627402382">

**Comment by snuup** [comment]
Thx, 

About (2) I do not have a workoad in Visual Studio, uninstalling that worked.  
About (1) :

    C:\Windows\system32>dotnet workload search

    Workload ID                 Description
    ------------------------------------------------------------------------------------
    android                     .NET SDK Workload for building Android applications.
    aspire                      .NET Aspire SDK
    ...

    C:\Windows\system32>dotnet workload uninstall aspire

    Workload uninstallation failed: Couldn't find workload ID(s): aspire

What is the problem here?

**Comment by clarkebowers** [comment]
I do not have this workload but aspire project types appear.

PS C:\WINDOWS\system32> dotnet workload uninstall aspire

Workload uninstallation failed: Couldn't find workload ID(s): aspire
PS C:\WINDOWS\system32> dotnet workload search aspire

Workload ID      Description
----------------------------

PS C:\WINDOWS\system32>

**Reply by davidfowl** [reply]
The dotnet workload is long gone. You should be able to remove the templates using the second approach *and* using the dotnet project templates:

```
dotnet new uninstall Aspire.ProjectTemplates
```

**Reply by clarkebowers** [reply]
No luck.

**********************************************************************
** Visual Studio 2026 Developer PowerShell v18.7.3
** Copyright (c) 2026 Microsoft Corporation
**********************************************************************
PS C:\Users\Clarke\Source\Repos\LauraWebTunes> dotnet new uninstall Aspire.ProjectTemplates
The template package 'Aspire.ProjectTemplates' is not found.
To list installed template packages use:
   dotnet new uninstall


For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
PS C:\Users\Clarke\Source\Repos\LauraWebTunes> dotnet new uninstall
Currently installed items:
(No Items)
PS C:\Users\Clarke\Source\Repos\LauraWebTunes>
